### PR TITLE
Archer RSA - fixed an issue where empty argument caused a crash and added UT

### PR DESCRIPTION
**Pull request by abaumgarten**
## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/29833

## Description
Fixed an issue where missing RequestedObject in the response caused a crash. 
Added an UT.

## Minimum version of Demisto
- [x] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2_test.py [hunk 1]
```diff
@@ -132,22 +135,48 @@
...
+
+GET_RESPONSE_SUCCESSFUL_JSON = {"IsSuccessful": True, "RequestedObject": {'SessionToken': 'session-id'}}
+
 
 def test_extract_from_xml():
     field_id = extract_from_xml(XML_FOR_TEST, 'Envelope.Body.GetValueListForField.fieldId')
     assert field_id == '6969'
 
 
-def test_get_level_by_app_id(requests_mock):
```
Comment (adi88d) [inline Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2_test.py:141]: please test the login in different method

File: Packs/ArcherRSA/ReleaseNotes/1_1_2.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### RSA Archer v2
+- Fixed an issue where receiving an empty HTTP response caused an error.
```
Comment (adi88d) [inline Packs/ArcherRSA/ReleaseNotes/1_1_2.md:4]:
```suggestion
- Improve error handling in login process.
```

**Review by adi88d — CHANGES_REQUESTED** [review]
nice!

File: Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2_test.py [hunk 3]
```diff
@@ -132,22 +135,48 @@
...
-    assert levels == GET_LEVELS_BY_APP
+    if is_successful:
+        levels = client.get_level_by_app_id('1')
+        assert levels == GET_LEVELS_BY_APP
+    else:
+        with pytest.raises(SystemExit) as e:
+            # return_error reached
+            client.get_level_by_app_id('1')
+        if not e:
+            assert False
```
Comment (guykeller) [inline Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2_test.py:179]: `assert False` is not a good practice
let's make it clear what the test is testing

**Review by guykeller — CHANGES_REQUESTED** [review]

File: Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2_test.py (hunk 1 above)
Comment (abaumgarten) [reply Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2_test.py:141]: Done

File: Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2_test.py (hunk 3 above)
Comment (abaumgarten) [reply Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2_test.py]: Done

**Comment by guykeller** [comment]
Looks good to me.
However, @abaumgarten please wait for the final call by @adi88d

**Review by guykeller — APPROVED** [review]

**Review by adi88d — APPROVED** [review]

**Pull request merged**